### DISCUSSION TITLE
How to remove node definition, which is referenced by included .dtsi, from the overlay

### DISCUSSION THREAD

**Discussion by yashi** [zephyrproject-rtos/zephyr#35214] [Q&A] [answered]
I'm trying to create a custom board based on Arty A7 in the zephyr tree, which has the following dtsi.  Is there any way to disable or removed referenced nodes, itcm and dtcm, from the overlay?

`boards/arm/arty/dts/arty_a7_arm_designstart.dtsi`
```
chosen {
		  :
    zephyr,flash = &itcm;
    zephyr,sram = &dtcm;
}
```

`myboard.dts`
```
chosen {
    zephyr,flash = <disable?>
}

soc {
    /* I want to remove the following two node because I don't use it. */
    itcm: memory@.... {
     :
    }
    dtcm: memory@... {
      :
    }
```

Because `arty_a7_arm_designstart.dtsi` assumes both itcm and dtcm is provided by the overaly by using `chosen`, I'm assuming you can't remove them from the overlay.  Or, is there any way to remove the reference to itcm and dtcm from the overaly?  Like an empty phandle?

I can use `status = "disabled"` in both node but my question isn't that.  I'd like to remove the definition of them from the overlay.

My only idea is to move `chosen` from `arty_a7_arm_designstart.dtsi` to including `.dts`.  But I'm sure there is a reason why they are there.

EDIT:
It seems like `<>` works.
```
chosen {
    zephyr,flash = <>
}
```

**Comment by henrikbrixandersen** [comment, marked as answer]
You can use `/delete-property/` for deleting devicetree properties and `/delete-node/` for deleting devicetree nodes from an overlay: https://docs.zephyrproject.org/latest/guides/dts/howtos.html#use-devicetree-overlays

**Reply by hjhhskm** [reply]
/delete-node/ can't use in overlay

**Reply by henrikbrixandersen** [reply]
> /delete-node/ can't use in overlay

Why not?

**Reply by cdwilson** [reply]
> /delete-node/ can't use in overlay

For anybody coming across this in the future, I initially tried to delete a node using something like this in my overlay (which did not work):

```
&spi3 {
    cs-gpios = <&feather_header 15 GPIO_ACTIVE_LOW>;

    /delete-node/ w25q32jv;
```

However, this does work:

```
/delete-node/ &w25q32jv;

&spi3 {
	cs-gpios = <&feather_header 15 GPIO_ACTIVE_LOW>;
```

**Reply by cdwilson** [reply]
See https://github.com/zephyrproject-rtos/zephyr/discussions/67228#discussioncomment-8039133 for some additional examples.
